lib/ruby_lsp/listeners/completion.rb

```diff
@@ -322,11 +324,13 @@ def complete_require_relative(node)
       sig { params(node: Prism::CallNode, name: String).void }
       def complete_methods(node, name)
         # If the node has a receiver, then we don't need to provide local nor keyword completions
-        if !@global_state.has_type_checker && !node.receiver
+        if @sorbet_level == Document::SorbetLevel::Disabled && !node.receiver
           add_local_completions(node, name)
           add_keyword_completions(node, name)
         end
 
+        return if @sorbet_level != Document::SorbetLevel::Disabled && self_receiver?(node)
```

Comment:
In `typed: true` files or above, Sorbet can always provide completion for methods invoked directly on `self`, regardless of whether they are annotated or not.

Reply:
Can the context be capture in a comment above this line?